src/feruchemy/java/leaf/cosmere/feruchemy/client/utils/FeruchemyChargeThread.java

```diff
@@ -73,83 +73,95 @@ public void run() {
 
         while (!isStopping)
         {
-            // no serverside action, unloaded levels, or non-existent players allowed >:(
-            if (mc.level == null || mc.player == null || !mc.level.isClientSide)
+            try
             {
-                stop();
-                break;
-            }
+                // no serverside action, unloaded levels, or non-existent players allowed >:(
+                if (mc.level == null || mc.player == null || !mc.level.isClientSide)
+                {
+                    break;
+                }
 
-            if (mc.player.tickCount % 2 != 0)
-            {
-                try {
-                    Thread.sleep(50);       // 20 ticks per 1000ms, 1000/20 = 50, rest for 1 tick
-                } catch (InterruptedException e) {
-                    throw new RuntimeException(e);
+                if (mc.player.tickCount % 2 != 0)
+                {
+                    try
+                    {
+                        Thread.sleep(50);       // 20 ticks per 1000ms, 1000/20 = 50, rest for 1 tick
+                    }
+                    catch (InterruptedException e)
+                    {
+                        throw new RuntimeException(e);
```

Comment:
if an exception is thrown here, what happens? The object wont have set itself to isStopping, right?

Reply:
I don't actually remember, and I don't know why that throw is there. It just seemed important since it's so different from the other threads that I didn't touch it. I think I'll just get rid of it though